libraries/AP_Motors/AP_Motors_Class.h

```diff
@@ -64,6 +101,47 @@ class AP_Motors {
         MOTOR_FRAME_TYPE_BF_X_REV = 18, // X frame, betaflight ordering, reversed motors
     };
 
+    // return string corresponding to frame_type
+    static const char* get_type_string(uint16_t frame_type)
+    {
+        switch((AP_Motors::motor_frame_type)frame_type) {
+            case MOTOR_FRAME_TYPE_PLUS:
+                return "PLUS";
```

Comment:
These strings should exist on the classes themselves, rather than in a switch statement here.  So:

```
pbarker@bluebottle:~/rc/ardupilot(pr/decode-watchdog)$ mavproxy.py --help | lesspbarker@bluebottle:~/rc/ardupilot(pr/decode-watchdog)$ git diff
diff --git a/libraries/AP_Motors/AP_MotorsSingle.h b/libraries/AP_Motors/AP_MotorsSingle.h
index 5276c25d74..cadf9e05b6 100644
--- a/libraries/AP_Motors/AP_MotorsSingle.h
+++ b/libraries/AP_Motors/AP_MotorsSingle.h
@@ -31,6 +31,8 @@ public:
     // init
     void                init(motor_frame_class frame_class, motor_frame_type frame_type) override;
 
+    const char *name() const override { return "SINGLE"; }
+
     // set frame class (i.e. quad, hexa, heli) and type (i.e. x, plus)
     void set_frame_class_and_type(motor_frame_class frame_class, motor_frame_type frame_type) override;
 
pbarker@bluebottle:~/rc/ardupilot(pr/decode-watchdog)$ 
```

Reply:
sorry, I don't understand; frame_type strings should be handled differently than frame_class?

Reply:
Both should be retrievable in the same way - as method calls on the motors class.

Presumably MotorsMatrix would need to switch() on `_last_frame_class` and `_last_frame_type`

I'm guessing that `frame_name()` might return nullptr for in the base class?

Reply:
I see; didn't notice that frame_class and type were attributes.

Reply:
why not move _last_frame_class and _last_frame_type up to the base class and require init() and set_frame_class_and_type() to maintain them?

Reply:
Try it and see?  I think the answer might be, "because only MotorsMatrix cares about frame type" - but that should become evident.

Most of the classes should be returning a static string for the class - they are what they are, and we shouldn't change the string being returned from motors based on the current configuration, rather returning what's allocated and/or active.